src/commands/createFunction/createFunction.ts

```diff
@@ -146,8 +151,8 @@ export async function createFunction(
         const setting: ConfigSetting | undefined = await templateData.getSetting(runtime, template.functionConfig.inBindingType, settingName);
         if (setting) {
             let settingValue: string | undefined;
-            if (functionSettings.length > 0) {
-                settingValue = functionSettings.shift();
+            if (functionSettings[settingName.toLowerCase()]) {
```

Comment:
Are empty string or zero valid setting values?

Reply:
Depends on the language/setting. I'll change this - ultimately as an api I think it would be better for us to use what was passed in (even if it's an empty string that's not valid)